### PR TITLE
feat: Increase verbosity of duplicate column error message

### LINKED ISSUE #11812: Can't use max_horizontal and min_horizontal in same with_columns in 0.19.9
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({"values":[1,2.0]})
df.with_columns(
        pl.max_horizontal(pl.col(pl.Float64)),
        pl.min_horizontal(pl.col(pl.Float64)),
    )
```

### Log output

```shell
ComputeError: The name: 'values' passed to `LazyFrame.with_columns` is duplicate

Error originated just after this operation:
DF ["values"]; PROJECT */1 COLUMNS; SELECTION: "None"
```

### Issue description

This worked in 0.19.8 with an output having max and min columns. Works if I replace e.g. pl.max_horizontal with pl.any_horizontal. Just doesn't work for max_h and min_h together

### Expected behavior

Columns called max and min added

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.9
Index type:          UInt32
Platform:            macOS-12.6.7-x86_64-i386-64bit
Python:              3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:54) [Clang 13.0.1 ]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         2.0.0
connectorx:          0.3.1
deltalake:           0.9.0
fsspec:              2023.5.0
gevent:              <not installed>
matplotlib:          3.7.1
numpy:               1.24.3
openpyxl:            <not installed>
pandas:              2.0.1
pyarrow:             13.0.0
pydantic:            1.10.7
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          2.0.12
xlsx2csv:            0.8.1
xlsxwriter:          3.0.3```

</details>

### LINKED ISSUE #14458: Increase verbosity of duplicate column message in `select`/`with_columns`
### Description

Ref https://github.com/pola-rs/polars/issues/11812#issuecomment-1768409858

Current error messages:

**select**

```python
import polars as pl

df = pl.DataFrame({"values": [1.0, 2.0]})
df.select(
    pl.max_horizontal(pl.col(pl.Float64)),
    pl.min_horizontal(pl.col(pl.Float64)),
)
```
```
polars.exceptions.DuplicateError: column with name 'values' has more than one occurrences
```

**with_columns**

```python
import polars as pl

df = pl.DataFrame({"values": [1.0, 2.0]})
df.with_columns(
    pl.max_horizontal(pl.col(pl.Float64)),
    pl.min_horizontal(pl.col(pl.Float64)),
)
```
```
polars.exceptions.ComputeError: The name: 'values' passed to `LazyFrame.with_columns` is duplicate

Error originated just after this operation:
DF ["values"]; PROJECT */1 COLUMNS; SELECTION: "None"
```

**Desired behavior**

Both should return a `DuplicateError` with a helpful error message like:

```
polars.exceptions.DuplicateError: the name 'values' passed to `LazyFrame.with_columns` is duplicate

Hint: Try renaming your expressions using `alias`.
```

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #14458

I notice that none of our rust errors are multi-line, is this okay here?

### CODE REVIEW

File: crates/polars-error/src/lib.rs [hunk 1]
```diff
@@ -247,7 +247,13 @@ Alternatively, if the performance cost is acceptable, you could just set:
 on startup."#.trim_start())
     };
     (duplicate = $name:expr) => {
-        polars_err!(Duplicate: "column with name '{}' has more than one occurrences", $name)
+        polars_err!(
+            Duplicate: "column '{}' is duplicate.\n\n\
```
Comment (ritchie46) [inline crates/polars-error/src/lib.rs:251]: I don't think we should have this message for all duplicate related errors. Only in `select` and `with_columns`.

**Review by ritchie46 — APPROVED** [review]

File: crates/polars-error/src/lib.rs (hunk 1 above)
Comment (mcrumiller) [reply crates/polars-error/src/lib.rs]: No problem--I think I fixed that with latest commit.

**Comment by mcrumiller** [comment]
I cannot replicate the hypothesis error. It appears it was failing in this case:

```
import polars as pl
from datetime import date, datetime

pl.DataFrame({
    "col0": pl.Series([0, 0], dtype=pl.UInt16),
    "col1": pl.Series([0, 0], dtype=pl.UInt16),
    "col2": pl.Series([datetime(2000, 1, 1), datetime(2000, 1, 1)], dtype=pl.Datetime("ns")),
    "col3": pl.Series([0, 0], dtype=pl.UInt16),
    "col4": pl.Series([0, 0], dtype=pl.UInt8),
    "col5": pl.Series([None, None], dtype=pl.Date),
    "col6": pl.Series([None, None], dtype=pl.Categorical),
}).null_count()
```
```
shape: (1, 7)
┌──────┬──────┬──────┬──────┬──────┬──────┬──────┐
│ col0 ┆ col1 ┆ col2 ┆ col3 ┆ col4 ┆ col5 ┆ col6 │
│ ---  ┆ ---  ┆ ---  ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ u32  ┆ u32  ┆ u32  ┆ u32  ┆ u32  ┆ u32  ┆ u32  │
╞══════╪══════╪══════╪══════╪══════╪══════╪══════╡
│ 0    ┆ 0    ┆ 0    ┆ 0    ┆ 0    ┆ 2    ┆ 2    │
└──────┴──────┴──────┴──────┴──────┴──────┴──────┘
```
This should pass the assertion test, no clue why it's reporting 2. I installed the same hypothesis version and set the replication decorator, but it fails to run on my machine.

**Comment by alexander-beedie** [comment]
> I cannot replicate the hypothesis error.

It's this: https://github.com/pola-rs/polars/issues/11910

**Comment by mcrumiller** [comment]
Thanks Alex. Once that gets resolved I'll pull those changes in here.

**Comment by mcrumiller** [comment]
@ritchie46 this one is now good to go.

File: crates/polars-core/src/frame/mod.rs [hunk 2]
```diff
@@ -1518,7 +1518,14 @@ impl DataFrame {
         let mut names = PlHashSet::with_capacity(cols.len());
         for name in cols {
             if !names.insert(name.as_str()) {
-                polars_bail!(duplicate = name);
+                let msg = format!(
```
Comment (ritchie46) [inline crates/polars-core/src/frame/mod.rs:1521]: This is still to broad IMO. I think this should be here: 

https://github.com/pola-rs/polars/blob/a28b9220da58e038adeaf63d46225fa96d05f161/crates/polars-lazy/src/physical_plan/executors/projection_utils.rs#L239

**Review by ritchie46 — CHANGES_REQUESTED** [review]

**Comment by stinodego** [comment]
@mcrumiller I opened a new issue specifically for the improvement in this PR. Could you address Ritchie's comment and finish this improvement?

**Comment by mcrumiller** [comment]
@stinodego I think I accomplished what he suggested, let me know if it still needs work and/or you have any suggestions.

**Comment by mcrumiller** [comment]
Edit: fixed. Got some failures due to the new CI issue.

**Comment by mcrumiller** [comment]
```python
>>> import polars as pl
>>> df = pl.DataFrame({"a": [1], "b": [1]})
>>> df.with_columns(pl.all().alias("a"))
polars.exceptions.ComputeError: the name: 'a' passed to `LazyFrame.with_columns` is duplicate

It's possible that multiple expressions are returning the same default column name. If this is the case, try renaming the columns with `.alias("new_name")` to avoid duplicate column names.
>>> df.lazy().with_columns(pl.all().alias("a")).collect()
polars.exceptions.ComputeError: the name: 'a' passed to `LazyFrame.with_columns` is duplicate

It's possible that multiple expressions are returning the same default column name. If this is the case, try renaming the columns with `.alias("new_name")` to avoid duplicate column names.
```

**Comment by stinodego** [comment]
> Should we add newlines for our long error messages, or let the terminal wrap them?

Terminal should wrap them.

File: crates/polars-lazy/src/physical_plan/executors/projection_utils.rs [hunk 3]
```diff
@@ -284,7 +284,17 @@ pub(super) fn check_expand_literals(
                 all_equal_len = false;
             }
             let name = s.name();
-            polars_ensure!(names.insert(name), duplicate = name);
+
+            if !names.insert(name) {
+                let msg = format!(
+                    "the name: '{}' passed to `select` is duplicate\n\n\
```
[the remark this answers is not in the document]
Comment (mcrumiller) [reply crates/polars-lazy/src/physical_plan/executors/projection_utils.rs]: Indeed; fixed.

**Comment by mcrumiller** [comment]
I'll rebase to fix the CI tests.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**